GoogleForCreators/web-stories-wp
# Don’t run db upgrader for fresh sites

## Context



See #12782 for details/explanation

## Summary



* Tweak `Database_Upgrader::run_upgrades()` so that it does not run upgrade routines for fresh installs.
  Instead, sets the database version to the latest one and bails.
* Remove `Yoast_Reindex_Stories` upgrade routine
  Yoast asked whether we could remove it so that they could safely deprecate the classes we used. Given that we only needed this one time when we changed permalink structure, and now we don't run this code anymore for new sites, this should be fine.
* Adds new methods to `Media_Source_Taxonomy` to insert missing terms upon plugin activation or site initialization.
* Ensures that `Story_Post_Type` is a dependency of `Capabilities`, so that when we want to add the missing capabilities to roles the post type is definitely already registered
* Implements significant bug fix in `ServiceBasedPlugin` to ensure that lazily evaluated services (like the `Capabilities` service) can be used for things like plugin activation

## Relevant Technical Choices



* Adds a new `Media_Source` pseudo enum to make code a bit more DRY and less error prone.

## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Install plugin on a fresh site (e.g. via tastewp.com)
2. Activate plugin
3. Verify that you can access the Web Stories dashboard and editor
4. Repeat with a Multisite setup (tastewp.com makes this easy)
5. Create a new sub-site on the Multisite, add a user to it
6. Log in as that user and ensure you can still access the dashboard and editor


## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #12782
